http_backend.go

```diff
@@ -139,10 +139,10 @@ func (h *httpBackend) Cache(request *http.Request, bodySize int, cacheDir string
 	if err != nil {
 		return resp, err
 	}
-	defer file.Close()
 	if err := gob.NewEncoder(file).Encode(resp); err != nil {
 		return resp, err
```

Comment:
please close the file before the return statement.

Reply:
Good catch.

 I just added another file.close() before the error return as this seems the most simple/readable.